pink/rondis/string/db_operations.cc

```diff
@@ -614,3 +614,173 @@ int rondb_get_rondb_key(const NdbDictionary::Table *tab,
     }
     return 0;
 }
+
+#define RONDB_INSERT 2
+#define RONDB_UPDATE 1
+#define REG0 0
+#define REG1 1
+#define REG2 2
+#define REG3 3
+#define REG4 4
+#define REG5 5
+#define REG6 6
+#define REG7 7
+#define LABEL0 0
+#define LABEL1 1
+void incr_key_row(std::string *response,
+                  Ndb *ndb,
+                  const NdbDictionary::Table *tab,
+                  NdbTransaction *trans,
+                  struct key_table *key_row) {
+
+    const NdbDictionary::Column *value_start_col = tab->getColumn(KEY_TABLE_COL_value_start);
+    const NdbDictionary::Column *tot_value_len_col = tab->getColumn(KEY_TABLE_COL_tot_value_len);
+    const NdbDictionary::Column *rondb_key_col = tab->getColumn(KEY_TABLE_COL_rondb_key);
+
+    NdbOperation::OperationOptions opts;
+    std::memset(&opts, 0, sizeof(opts));
+    /**
+     * The mask specifies which columns is to be updated after the interpreter
+     * has finished. The values are set in the key_row.
+     * We have 7 columns, we will update tot_value_len in interpreter, same with
+     * value_start.
+     *
+     * The rest, redis_key, rondb_key, value_data_type, num_rows and expiry_date
+     * are updated through final update.
+     */
+
+    const Uint32 mask = 0x55;
```

Comment:
Why not use a base-10 integer?

Reply:
Because if I write 85 it gives no easy translation to which attribute ids that should be written.
0x55 is binary 0101 0101 thus I can conclude that we will write attribute 0, 2, 4 and 6 in the Update
section, the values for those columns will come from key_tab struct.

We will not set column 1 (== rondb_key) which means it will retain its value (it should always be NULL for INCR)
We will not set column 3 (tot_value_len) since it is set by the interpreted program
We will not set column 5 (value_start) since it is set by the interpreted program.

Reply:
Aha okay, that makes more sense now, thanks :)